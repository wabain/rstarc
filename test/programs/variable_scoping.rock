FuncA takes X
    FuncB takes Y
        If Z isn't mysterious
          Put Y plus Z into Z
        Else
          Put Y into Z
        (endif)
    (endfunc)
    If X isn't nothing
        If Z isn't mysterious
            Put X plus Z into Z
        Else
            Put X into Z
        (endif)
    (endif)
    Give back FuncB
(endfunc)
Put FuncA taking nothing into FB
Say Z (expect mysterious)
Put FB taking 100 into Dummy
Say Z (expect 100)
Put FuncA taking 200 into Dummy
Say Z (expect 300)
Put 300 plus Z into Z
Say Z (expect 600)