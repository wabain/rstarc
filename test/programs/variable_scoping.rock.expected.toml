[tests]
enabled = ["run"]

["run.exec"]
returncode = "1"
stderr = """
error: using non-local variable 'Z' in compiled code is not supported.

Try executing your program using 'rstarc run --interpret <program>' instead

3:          If Z is not mysterious
               ^
"""

["run.interpret"]
returncode = "0"
stdout = """
mysterious
100
300
600
"""
